ykcompile/src/lib.rs

```diff
@@ -144,12 +147,42 @@ pub struct TraceCompiler<TT> {
     variable_location_map: HashMap<Local, Location>,
     /// Local referencing the input arguments to the trace.
     trace_inputs_local: Option<Local>,
+    /// Local decls of the tir trace.
+    local_decls: HashMap<Local, LocalDecl>,
     /// Stack builder for allocating objects on the stack.
     stack_builder: StackBuilder,
     _pd: PhantomData<TT>,
 }
 
 impl<TT> TraceCompiler<TT> {
+    fn can_live_in_register(tyid: &TypeId) -> bool {
+        // FIXME: optimisation: small structs and tuples etc. could actually live in a register.
+        let ty = SIR.ty(tyid);
+        match ty {
+            Ty::UnsignedInt(ui) => match ui {
```

Comment:
Since these are all true and we won't see any other types here, should we just squash this down to a single `true` statement?

Reply:
Good point.